diff --git a/src/physics/ncar_ccpp/schemes/conservation_adjust/check_energy/check_energy_gmean/check_energy_gmean.F90 b/src/physics/ncar_ccpp/schemes/conservation_adjust/check_energy/check_energy_gmean/check_energy_gmean.F90
index 69816ae..3e9c6c4 100644
--- a/src/physics/ncar_ccpp/schemes/conservation_adjust/check_energy/check_energy_gmean/check_energy_gmean.F90
+++ b/src/physics/ncar_ccpp/schemes/conservation_adjust/check_energy/check_energy_gmean/check_energy_gmean.F90
@@ -1,5 +1,6 @@
 module check_energy_gmean
   use ccpp_kinds, only: kind_phys
+  use physics_grid, only: pcols => columns_on_task

   implicit none
   private
@@ -49,10 +50,13 @@ contains
     integer,            intent(out)   :: errflg         ! error flag

     ! Local variables
-    real(kind_phys) :: te(ncol, 4)                      ! total energy of input/output states (copy)
+    real(kind_phys) :: te(pcols, 4)                     ! total energy of input/output states (copy)
     real(kind_phys) :: te_glob(4)                       ! global means of total energy

     errmsg = ''
     errflg = 0
+
+    ! gmean operates on pcols rows, including inactive padding columns.
+    te = 0._kind_phys

     ! Copy total energy out of input and output states.
