diff --git a/src/physics/ncar_ccpp/schemes/musica/micm/musica_ccpp_micm.F90 b/src/physics/ncar_ccpp/schemes/musica/micm/musica_ccpp_micm.F90
--- a/src/physics/ncar_ccpp/schemes/musica/micm/musica_ccpp_micm.F90
+++ b/src/physics/ncar_ccpp/schemes/musica/micm/musica_ccpp_micm.F90
@@ -127,7 +127,9 @@
       call constituent_props(species_index)%instantiate( &
         std_name = species_name, &
         long_name = species_name, &
+#ifdef PYCAM_CCPP_DIAG_NAME
         diag_name = species_name, &
+#endif
         units = 'kg kg-1', &
         vertical_dim = 'vertical_layer_dimension', &
         default_value = default_value, &
diff --git a/src/physics/ncar_ccpp/schemes/musica/tuvx/musica_ccpp_tuvx_load_species.F90 b/src/physics/ncar_ccpp/schemes/musica/tuvx/musica_ccpp_tuvx_load_species.F90
--- a/src/physics/ncar_ccpp/schemes/musica/tuvx/musica_ccpp_tuvx_load_species.F90
+++ b/src/physics/ncar_ccpp/schemes/musica/tuvx/musica_ccpp_tuvx_load_species.F90
@@ -74,7 +74,9 @@
     call temp_constituent_props(i_new)%instantiate( &
       std_name = CLOUD_LIQUID_WATER_CONTENT_LABEL, &
       long_name = CLOUD_LIQUID_WATER_CONTENT_LONG_NAME, &
+#ifdef PYCAM_CCPP_DIAG_NAME
       diag_name = CLOUD_LIQUID_WATER_CONTENT_LABEL, &
+#endif
       units = CLOUD_LIQUID_WATER_CONTENT_UNITS, &
       vertical_dim = "vertical_layer_dimension", &
       default_value = 0.0_kind_phys, &
@@ -110,7 +112,9 @@
       call temp_constituent_props(i_new)%instantiate( &
         std_name = DRY_AIR_LABEL, &
         long_name = DRY_AIR_LABEL, &
+#ifdef PYCAM_CCPP_DIAG_NAME
         diag_name = DRY_AIR_LABEL, &
+#endif
         units = 'kg kg-1', &
         vertical_dim = "vertical_layer_dimension", &
         default_value = 0.0_kind_phys, &
@@ -127,7 +131,9 @@
       call temp_constituent_props(i_new)%instantiate( &
         std_name = O2_LABEL, &
         long_name = O2_LABEL, &
+#ifdef PYCAM_CCPP_DIAG_NAME
         diag_name = O2_LABEL, &
+#endif
         units = 'kg kg-1', &
         vertical_dim = "vertical_layer_dimension", &
         default_value = 0.0_kind_phys, &
@@ -144,7 +150,9 @@
       call temp_constituent_props(i_new)%instantiate( &
         std_name = O3_LABEL, &
         long_name = O3_LABEL, &
+#ifdef PYCAM_CCPP_DIAG_NAME
         diag_name = O3_LABEL, &
+#endif
         units = 'kg kg-1', &
         vertical_dim = "vertical_layer_dimension", &
         default_value = 0.0_kind_phys, &
